pkg/models/devops/devops.go

```diff
@@ -175,6 +175,9 @@ func (d devopsOperator) ListDevOpsProject(workspace string, limit, offset int) (
 		limit = len(result) - offset
 	}
 	items = result[offset : offset+limit]
+	if items == nil {
```

Comment:
why items is nil since you've already initialized in line https://github.com/kubesphere/kubesphere/pull/2193/files#diff-2c85fb739a144b30d0a086122b84ab75R168

Reply:
When return nil , It will work.  But , this function return {Items:nil, TotalItems:0}.

Reply:
What I mean is `items` can't be nil, since you've already initialized above,  statement `items = result[offset : offset+limit]` also can't lead to a nil `items`. So if this function returns a nil `items`, I think maybe somewhere else.

Reply:
when result is nil ，item wil get a nil value.

Reply:
If result is nil, what do you think will happen to this statement
```
items = result[offset: offset+limit]
```

Reply:
nill[0:0] is ok.

Reply:
OK，just tried, it still passes when result is nil, that's wired